vehicle/Sedan.h

```diff
@@ -0,0 +1,17 @@
+#ifndef SEDAN_H
+#define SEDAN_H
+
+#include "ICar.h"
+
+template <typename EngineType, typename GearType>
```

Comment:
same here: why not simply use interface?

Reply:
template removed